### PR TITLE
[JUJU-3232] Rephrase the error message in caas provisioner worker for less confus…

### PR DESCRIPTION
**Pull request by ycliuhw**
Rephrase the error message for less confusion.

## Bug reference

https://bugs.launchpad.net/juju/+bug/2009762

### CODE REVIEW

File: worker/caasapplicationprovisioner/application.go [hunk 1]
```diff
@@ -144,13 +144,13 @@ func (a *appWorker) loop() error {
 
 		exists, err := a.broker.OperatorExists(a.name)
 		if err != nil {
-			return errors.Trace(err)
+			return errors.Annotatef(err, "checking if %q has an operator pod due to an upgrade-charm from a pod-spec charm to a sidecar charm", a.name)
```
Comment (wallyworld) [inline worker/caasapplicationprovisioner/application.go:147]: "... due to upgrading the charm from a pod-spec charm to ..."

File: worker/caasapplicationprovisioner/application.go [hunk 2]
```diff
@@ -144,13 +144,13 @@ func (a *appWorker) loop() error {
...
 		}
 		if !exists.Exists {
 			break
 		}
 
-		a.logger.Infof("deleting workload and operator pods for application %q", a.name)
+		a.logger.Infof("app %q has just been upgraded from a podspec charm to sidecar, now deletes workload and operator pods %q", a.name)
```
Comment (wallyworld) [inline worker/caasapplicationprovisioner/application.go:153]: " .. sidecar, now deleting workload and ...."

**Review by wallyworld — APPROVED** [review]
Just some typos

File: worker/caasapplicationprovisioner/application.go (hunk 1 above)
Comment (ycliuhw) [reply worker/caasapplicationprovisioner/application.go:147]: hmm, this was copied from the comment above..

**Pull request merged**